log/log.go

```diff
@@ -141,3 +141,28 @@ func IfErr(l Logger, err error) {
 		l.Log(Err, err)
 	}
 }
+
+// IfErrAndReturn is a shorthand that will log an error if err is not nil and returns the original err
+func IfErrAndReturn(l Logger, err error) error {
+	if err != nil {
+		l.Log(Err, err)
+	}
+	return err
+}
+
+// IfErrWithKeys logs an error with the supplied logger if it is not nil.
+// The error will be appended to the end of the supplied keys/messages
+func IfErrWithKeys(l Logger, err error, intf ...interface{}) {
```

Comment:
If we use this a lot in the code then won't it have performance impact with append. Can we do a benchmark on this just to get an idea?

Reply:
What would you like for me to compare it to?  IfErr?